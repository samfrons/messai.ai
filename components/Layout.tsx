import { ReactNode } from 'react'

interface LayoutProps {
  children: ReactNode
}

export default function Layout({ children }: LayoutProps) {
  return (
    <div className="min-h-screen bg-gray-50">
      <header className="bg-white shadow-sm border-b">
        <div className="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8">
          <div className="flex justify-between items-center py-6">
            <h1 className="text-xl font-semibold text-gray-900">
              MESSAi Platform
            </h1>
            <nav className="flex space-x-4">
              <a href="/" className="text-gray-600 hover:text-gray-900">
                Home
              </a>
            </nav>
          </div>
        </div>
      </header>
      <main className="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8 py-8">
        {children}
      </main>
    </div>
  )
}