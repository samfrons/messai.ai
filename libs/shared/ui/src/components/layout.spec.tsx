import type React from 'react';
import { render, screen } from '@messai/testing';
import Layout from './layout';

// Mock Next.js components
jest.mock('next/link', () => {
  return function MockLink({ children, href }: { children: React.ReactNode; href: string }) {
    return <a href={href}>{children}</a>;
  };
});

describe('Layout Component', () => {
  const mockChildren = <div data-testid="test-content">Test Content</div>;

  it('should render children content', () => {
    render(<Layout>{mockChildren}</Layout>);

    expect(screen.getByTestId('test-content')).toBeInTheDocument();
    expect(screen.getByText('Test Content')).toBeInTheDocument();
  });

  it('should have semantic HTML structure', () => {
    render(<Layout>{mockChildren}</Layout>);

    const main = screen.getByRole('main');
    expect(main).toBeInTheDocument();
    expect(main).toContainElement(screen.getByTestId('test-content'));
  });

  it('should apply base styling classes', () => {
    const { container } = render(<Layout>{mockChildren}</Layout>);

    // Find the actual layout div (it might be wrapped by test providers)
    const layoutDiv = container.querySelector('div.min-h-screen');
    expect(layoutDiv).toBeInTheDocument();
    expect(layoutDiv).toHaveClass('min-h-screen', 'bg-gray-50');
  });

  it('should be accessible', () => {
    render(<Layout>{mockChildren}</Layout>);

    // Check for proper heading hierarchy
    const mainContent = screen.getByRole('main');
    expect(mainContent).toBeInTheDocument();
  });

  it('should handle empty children gracefully', () => {
    render(<Layout>{null}</Layout>);

    const main = screen.getByRole('main');
    expect(main).toBeInTheDocument();
    expect(main).toBeEmptyDOMElement();
  });

  it('should handle multiple children', () => {
    render(
      <Layout>
        <div data-testid="child-1">Child 1</div>
        <div data-testid="child-2">Child 2</div>
      </Layout>
    );

    expect(screen.getByTestId('child-1')).toBeInTheDocument();
    expect(screen.getByTestId('child-2')).toBeInTheDocument();
  });

  it('should maintain consistent layout structure', () => {
    const { container } = render(<Layout>{mockChildren}</Layout>);

    // Verify the layout maintains consistent structure
    expect(container.firstChild).toMatchSnapshot();
  });
});
