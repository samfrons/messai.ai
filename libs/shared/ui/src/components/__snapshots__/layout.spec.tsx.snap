// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Component should maintain consistent layout structure 1`] = `
<div
  data-testid="test-wrapper"
>
  <div
    class="min-h-screen bg-gray-50"
  >
    <header
      class="bg-white shadow-sm border-b"
    >
      <div
        class="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8"
      >
        <div
          class="flex justify-between items-center py-6"
        >
          <h1
            class="text-xl font-semibold text-gray-900"
          >
            MESSAi Platform
          </h1>
          <nav
            class="flex space-x-4"
          >
            <a
              class="text-gray-600 hover:text-gray-900"
              href="/"
            >
              Home
            </a>
          </nav>
        </div>
      </div>
    </header>
    <main
      class="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8 py-8"
    >
      <div
        data-testid="test-content"
      >
        Test Content
      </div>
    </main>
  </div>
</div>
`;
